### PR TITLE
Disable "Execute" if automation is unavailable

### LINKED ISSUE #6825: Unavailable automation show execute button
## Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [x] I have tried a different browser to see if it is related to my browser.

## The problem

When a YAML automation is removed, but still in the entity registry, the entity can be removed.
However, it also still provides an "EXECUTE" button, which I did not expect to be here.

As the automation doesn't exist, it cannot be executed.

## Expected behavior

Not to have the "EXECUTE" button, or have it is in a disabled state.

## Steps to reproduce

1. Create a YAML automation with an unique ID in it.
2. After reloading/restarting, remove the YAML configuration
3. Reload/restart again
4. "EXECUTE" the removed automation

## Environment

- Home Assistant release with the issue: 0.115.0b0
- Last working Home Assistant release (if known): unknown
- Browser and browser version: not relevant
- Operating system: not relevant

## State of relevant entities

```yaml

```

## Problem-relevant configuration

```yaml

```

## Javascript errors shown in your browser console/inspector

```txt

```

### PR DESCRIPTION
**Pull request by spacegaier**
## Proposed change

Disable "Execute" button in automation picker and more-info dialog, if automation is in state "Unavailable".

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #6825
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/dialogs/more-info/controls/more-info-automation.ts [hunk 1]
```diff
@@ -34,7 +34,10 @@ class MoreInfoAutomation extends LitElement {
       </div>
 
       <div class="actions">
-        <mwc-button @click=${this.handleAction}>
+        <mwc-button
+          @click=${this.handleAction}
+          .disabled=${this.stateObj!.state === "unavailable"}
```
Comment (zsarnett) [inline src/dialogs/more-info/controls/more-info-automation.ts:39]: We should hide it instead

Comment (spacegaier) [reply src/dialogs/more-info/controls/more-info-automation.ts:39]: Is there an general HA best-practise how to do so?

Comment (zsarnett) [reply src/dialogs/more-info/controls/more-info-automation.ts:39]: Dont use css

Comment (zsarnett) [reply src/dialogs/more-info/controls/more-info-automation.ts:39]: `${condition ? true : false }`

Comment (zsarnett) [reply src/dialogs/more-info/controls/more-info-automation.ts:39]: ${state !== unavailable ? html`Button HTML` : "" }

File: src/dialogs/more-info/controls/more-info-automation.ts [hunk 2]
```diff
@@ -34,9 +34,11 @@ class MoreInfoAutomation extends LitElement {
       </div>
 
       <div class="actions">
-        <mwc-button @click=${this.handleAction}>
-          ${this.hass.localize("ui.card.automation.trigger")}
-        </mwc-button>
+        ${this.stateObj!.state !== "unavailable"
+          ? html` <mwc-button @click=${this.handleAction}>
```
Comment (zsarnett) [inline src/dialogs/more-info/controls/more-info-automation.ts:38]:
```suggestion
          ? html`<mwc-button @click=${this.handleAction}>
```

**Review by zsarnett — APPROVED** [review]

File: src/dialogs/more-info/controls/more-info-automation.ts (hunk 1 above)
Comment (bramkragten) [reply src/dialogs/more-info/controls/more-info-automation.ts]: Why hide and not disable?

Comment (spacegaier) [reply src/dialogs/more-info/controls/more-info-automation.ts]: I would be fine either way. I assumed performance reasons (less HTML to render)?
@zsarnett Can you chime in?

Comment (ludeeus) [reply src/dialogs/more-info/controls/more-info-automation.ts]: I think disabled tell more to the user than a button missing

Comment (bramkragten) [reply src/dialogs/more-info/controls/more-info-automation.ts]: We disable things when entities are unavailable in general, so we should do it here too.

Comment (zsarnett) [reply src/dialogs/more-info/controls/more-info-automation.ts]: Doesnt the text already say that?

Comment (zsarnett) [reply src/dialogs/more-info/controls/more-info-automation.ts]: Why show a button and text saying its unavailable?

Comment (bramkragten) [reply src/dialogs/more-info/controls/more-info-automation.ts]: So a user knows it normally would be available

Comment (spacegaier) [reply src/dialogs/more-info/controls/more-info-automation.ts:39]: I agree (e.g. the "enable" switch is also disabled and not hidden) so I changed this PR back to disabling of the "Execute" button.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**